File: ch2/2-4.c
#include <stdio.h>

void squeeze(char s1[], const char s2[]);

int main(void)
{
    char name[] = "Jonathan";
    printf("%s\n", name);
    squeeze(name, "Ja");
    printf("%s\n", name);
    return 0;
}

void squeeze(char s1[], const char s2[])
{
    int i, j, k;
    for (i = 0, k = 0; s1[i] != '\0'; i++) {
        int in_s2 = 0;
        for (j = 0; s2[j] != '\0'; j++) {
            if (s1[i] == s2[j]) {
                in_s2 = 1;
                break;
            }
        }
        if (in_s2 == 0) {
            s1[k++] = s1[i];
        }
    }
    s1[k] = '\0';
}

File: ch2/2-4.go
package main

import "fmt"

func main() {
	name := "Jonathan"
	fmt.Printf("%s\n", name)
	name = squeeze([]byte(name), "Ja")
	fmt.Printf("%s\n", name)
}

func squeeze(s1 []byte, s2 string) string {
	var i, j, k int
	for i, k = 0, 0; i < len(s1); i++ {
		var in bool
		for j = 0; j < len(s2); j++ {
			if rune(s1[i]) == rune(s2[j]) {
				in = true
				break
			}
		}
		if !in {
			s1[k] = s1[i]
			k++
		}
	}
	if len(s2) > 0 {
		s1 = s1[:(len(s1)-len(s2))-1]
	}
	return string(s1)
}
